resolwe/test/testcases/process.py

```diff
@@ -729,6 +750,127 @@ def assertFilesExist(self, obj, field_path):  # pylint: disable=invalid-name
+        :type obj: ~resolwe.flow.models.Data
+
+        :param str dir_path: path to the directory to compare
+
+        :param dict dir_struct: correct tree structure of the directory.
+            Dictionary keys are directory and file names with the correct nested
+            structure. Dictionary value associated with each directory is a new
+            dictionary which lists the content of the directory. Dictionary
+            value associated with each file name is ``None``
+
+        :param bool exact: if ``True`` tested directory structure must exactly
+            match `dir_struct`. If ``False`` `dir_struct` must be a partial
+            structure of the directory to compare
+
+        """
+        self.assertDirExists(obj, field_path)
+        field = dict_dot(obj.output, field_path)
+        dir_path = obj.location.get_path(filename=field['dir'])
+        self._assert_dir_structure(dir_path, dir_struct, exact)
+
+    def _assert_dir(self, dir_path, fn_correct, fail_on_funny=True):
+        """Compare directory `dir_path` to compressed directory `fn_correct`."""
+        correct_path = os.path.join(self.files_path, fn_correct)
+        if not os.path.isfile(correct_path):
+            with tarfile.open(correct_path, 'w:gz') as f:
+                for content in os.listdir(dir_path):
+                    f.add(os.path.join(dir_path, content), arcname=content)
+            self.fail(msg="Compressed output directory {} missing so it was created.".format(fn_correct))
+
+        if not tarfile.is_tarfile(correct_path):
+            self.fail(msg="{} is not a tar file.".format(fn_correct))
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            with tarfile.open(correct_path) as tar:
+                tar.extractall(temp_dir)
+            cmp = filecmp.dircmp(dir_path, temp_dir)
+            if cmp.left_only or cmp.right_only or cmp.diff_files or (fail_on_funny and cmp.funny_files):
+                self.fail(msg='Directory {} content mismatch: {}.'.format(fn_correct, cmp.report()))
+
+    def assertDir(self, obj, field_path, fn):  # pylint: disable=invalid-name
+        """Compare process output directory to correct compressed directory.
+
+        :param obj: object that includes the directory to compare
+        :type obj: ~resolwe.flow.models.Data
+
+        :param str field_path: path to
+            :class:`~resolwe.flow.models.Data` object's field with the
+            file name
+
+        :param str fn: file name (and relative path) of the correct compressed
+            directory to compare against. Path should be relative to the
+            ``tests/files`` directory of a Django application. Compressed
+            directory needs to be in ``tar.gz`` format.
+
+        """
+        self.assertDirExists(obj, field_path)
+        field = dict_dot(obj.output, field_path)
+        dir_path = obj.location.get_path(filename=field['dir'])
+        self._assert_dir(dir_path, fn)
```

Comment:
Is there any advantage of moving this to a separate method?

Reply:
I followed the same design as in the [file counterpart](https://github.com/genialis/resolwe/blob/master/resolwe/test/testcases/process.py#L587). I presume that this is needed to be able to [test this method](https://github.com/genialis/resolwe/blob/ca9c22bcc057accbb8d8999722e289c6664ed09b/resolwe/test/tests/test_testing_framework.py#L211) without explicitly running processes.